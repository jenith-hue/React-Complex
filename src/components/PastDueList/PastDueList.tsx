import React, { useState } from 'react';
import { Footer } from '../../common/Footer/Footer';
import { PrintLetterProvider } from '../../context/PrintLetter/PrintLetterProvider';
import { SubNavigation } from '../SubNavigation/SubNavigation';
import { Filters } from './Filters/Filters';
import { PastDueCustomerList } from './PastDueCustomerList/PastDueCustomerList';
import { PastDueFooter } from './PastDueCustomerList/PastDueFooter';

export const PasDueList = () => {
  const [isFilterOpen, setIsFilterOpen] = useState<boolean>(true);

  const toggleFilterPanel = () => {
    setIsFilterOpen(!isFilterOpen);
  };

  return (
    <>
      <SubNavigation onFilterClick={toggleFilterPanel} />
      <Filters
        isOpen={isFilterOpen}
        handleToggleFilterPanel={toggleFilterPanel}
      />
      <PastDueCustomerList />
      <Footer>
        <PrintLetterProvider>
          <PastDueFooter />
        </PrintLetterProvider>
      </Footer>
    </>
  );
};
