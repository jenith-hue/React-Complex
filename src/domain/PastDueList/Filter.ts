export interface DateRange {
  startDate: string;
  endDate: string;
}

export interface Filter {
  bestTimeToCall: string[];
  commitmentDateRange?: DateRange;
  route?: string;
  includeAutoPay: string;
  daysPastDue: string[];
  //customersWithCallActivity?: string;
  commitmentStatus: string[];
  language: string[];
  communicationType: string[];
  pastDueDateRange?: DateRange;
  customersWithActivity?: string;
}

export interface FilterPayload extends Filter {
  storeNumber: string;
}

export const defaultFilter = {
  bestTimeToCall: [],
  storeNumber: '',
  includeAutoPay: 'Y',
  daysPastDue: [],
  commitmentStatus: [],
  language: [],
  communicationType: [],
};
