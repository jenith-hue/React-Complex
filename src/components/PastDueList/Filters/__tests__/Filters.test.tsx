/* eslint-disable react/display-name */
/* eslint-disable sonarjs/no-duplicate-string */
/* eslint-disable @typescript-eslint/no-empty-function */

import React from 'react';
import renderer from 'react-test-renderer';

import { Filters } from '../Filters';
import * as Contexts from '../../../../context/PastDueListSearchCriteria/PastDueListSearchCriteriaProvider';
import { defaultFilter } from '../../../../domain/PastDueList/Filter';
interface PropWithChildren {
  children: JSX.Element;
}
jest.mock('@rentacenter/racstrap', () => ({
  ...jest.requireActual('@rentacenter/racstrap'),
  RACButton: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACButton" {...rest}>
      {children}
    </div>
  ),
  RACCard: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACCard" {...rest}>
      {children}
    </div>
  ),
  RACCheckBox: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACCheckBox" {...rest}>
      {children}
    </div>
  ),
  RACDateRangePicker: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACDateRangePicker" {...rest}>
      {children}
    </div>
  ),
  RACSelect: ({ children, ...rest }: PropWithChildren) => {
    return (
      <div data-testid="RACSelect" {...rest}>
        {children}
      </div>
    );
  },
  Divider: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Divider" {...rest}>
      {children}
    </div>
  ),
  Grid: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Grid" {...rest}>
      {children}
    </div>
  ),
  Typography: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Typography" {...rest}>
      {children}
    </div>
  ),
  makeStyles: () => () => ({
    filtersRoot: 'filtersRoot',
    row: 'row',
    card: 'card',
    cardBody: 'cardBody',
    column: 'column',
    multipleElementsInGrid: 'multipleElementsInGrid',
    multipleElementsInGridTitle: 'multipleElementsInGridTitle',
    filtersmultipleElementsInGridInputsRoot: 'multipleElementsInGridInputs',
    grid: 'grid',
    centerVertically: 'centerVertically',
    buttonContainer: 'buttonContainer',
    noPaddingLeft: 'noPaddingLeft',
    selectOptionsPaper: 'selectOptionsPaper',
  }),
}));
describe('Filters', () => {
  const valueFromEvent = 'Value from event';
  const event = {
    target: {
      value: valueFromEvent,
      checked: true,
    },
  };

  const mockSetSelectedDaysLateOption = jest.fn();
  const mockSetSelectedBestTimeToCallOption = jest.fn();
  const mockSetSelectedCommitmentStatusOption = jest.fn();
  const mockSetSelectedLanguageOption = jest.fn();
  const mockSetSelectedRouteOption = jest.fn();
  const mockSetSelectedCommunicationTypeOption = jest.fn();
  const mockOnDueDateFromOptionChanged = jest.fn();
  const mockOnDueDateToOptionChanged = jest.fn();
  const mockSetIsAutopayCustomersOption = jest.fn();
  const mockOnDueDateCheckboxChanged = jest.fn();
  const mockOnCommitmentDateFromOptionChanged = jest.fn();
  const mockOnCommitmentDateToOptionChanged = jest.fn();
  const mockOnCustomersWithActivityChanged = jest.fn();
  const mockOnClearSelectedOptions = jest.fn();

  beforeEach(() => {
    jest.spyOn(Contexts, 'useFilters').mockImplementationOnce(() => ({
      daysLateOptions: [
        {
          label: 'Days late 1',
          value: 'Days late 1',
          referenceCode: '1',
          displaySeq: 1,
          defaultValue: '1',
          description: '1',
        },
      ],
      bestTimeToCallOptions: [
        {
          label: 'Best time to call 1',
          value: 'Best time to call 1',
          referenceCode: '1',
          displaySeq: 1,
          defaultValue: '1',
          description: '1',
        },
      ],
      commitmentStatusOptions: [
        {
          label: 'Commitment status 1',
          value: 'Commitment status 1',
          referenceCode: '1',
          displaySeq: 1,
          defaultValue: '1',
          description: '1',
        },
      ],
      communicationTypeOptions: [
        {
          label: 'Communication type 1',
          value: 'Communication type 1',
          referenceCode: '1',
          displaySeq: 1,
          defaultValue: '1',
          description: '1',
        },
      ],
      languageOptions: [
        {
          label: 'Language option 1',
          value: 'Language option 1',
          referenceCode: '1',
          displaySeq: 1,
          defaultValue: '1',
          description: '1',
        },
      ],
      customersWithActivityOptions: [
        {
          label: 'Customers with  activity 1',
          value: 'Customers with activity 1',
          referenceCode: '1',
          displaySeq: 1,
          defaultValue: '1',
          description: '1',
        },
      ],
      customersWithCallActivityToolTip: [
        {
          label: 'Customer with and without activity',
          value: 'All',
        },
      ],

      routeOptions: [
        {
          label: 'Route 1',
          value: 'Route 1',
          routeCode: '1',
          routeDescription: '1',
        },
      ],
      selectedDaysLateOption: ['Days late 1'],
      selectedBestTimeToCallOption: ['Best time to call 1'],
      selectedCommitmentStatusOption: ['Commitment status 1'],
      selectedCommunicationTypeOption: ['Communication type 1'],
      selectedLanguageOption: ['Language option 1'],
      selectedCustomersWithActivityOption: 'Customers with no call activity 1',
      selectedRouteOption: 'Route 1',
      selectedCommitmentDateFromOption: 'Commitment date from 1',
      selectedCommitmentDateToOption: 'Commitment date to 1',
      selectedDueDateFromOption: 'Due date from 1',
      selectedDueDateToOption: 'Due date to 1',
      isAutopayCustomersOptionSelected: true,
      isAllCommitmentDateOptionSelected: true,
      isAllDueDateOptionSelected: true,
      filter: defaultFilter,
      hasRouteApiError: false,
      hasApiError: false,
      isLoading: false,
      isFetchRoutesLoading: false,
    }));

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    //@ts-ignore
    jest.spyOn(Contexts, 'useFiltersActions').mockImplementationOnce(() => ({
      onBestTimeToCallChanged: mockSetSelectedBestTimeToCallOption,
      onCommitmentStatusChanged: mockSetSelectedCommitmentStatusOption,
      onCommunicationTypeChanged: mockSetSelectedCommunicationTypeOption,
      onDaysLateOptionChanged: mockSetSelectedDaysLateOption,
      onLanguageChanged: mockSetSelectedLanguageOption,
      setSelectedRouteOption: mockSetSelectedRouteOption,
      setIsAutopayCustomersOption: mockSetIsAutopayCustomersOption,
      onDueDateFromOptionChanged: mockOnDueDateFromOptionChanged,
      onDueDateToOptionChanged: mockOnDueDateToOptionChanged,
      onDueDateCheckboxChanged: mockOnDueDateCheckboxChanged,
      onCommitmentDateFromOptionChanged: mockOnCommitmentDateToOptionChanged,
      onCommitmentDateToOptionChanged: mockOnCommitmentDateFromOptionChanged,
      onCustomersWithActivityChanged: mockOnCustomersWithActivityChanged,
      onClearSelectedOptions: mockOnClearSelectedOptions,
    }));
  });
  it('renders inputs with values from context', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('renders inputs with values from context, even when filters are not open', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen={false} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('calls correct callback for days late selector', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ inputLabel: 'Days late' })[0]
      .props.onChange(event);

    expect(mockSetSelectedDaysLateOption).toHaveBeenCalledTimes(1);
    expect(mockSetSelectedDaysLateOption).toHaveBeenCalledWith(valueFromEvent);
  });
  it('calls correct callback for best time to call selector', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ inputLabel: 'Best Time to Call' })[0]
      .props.onChange(event);

    expect(mockSetSelectedBestTimeToCallOption).toHaveBeenCalledTimes(1);
    expect(mockSetSelectedBestTimeToCallOption).toHaveBeenCalledWith(
      valueFromEvent
    );
  });
  it('calls correct callback for commitment status selector', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ inputLabel: 'Commitment Status' })[0]
      .props.onChange(event);

    expect(mockSetSelectedCommitmentStatusOption).toHaveBeenCalledTimes(1);
    expect(mockSetSelectedCommitmentStatusOption).toHaveBeenCalledWith(
      valueFromEvent
    );
  });
  it('calls correct callback for language selector', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ inputLabel: 'Language' })[0]
      .props.onChange(event);

    expect(mockSetSelectedLanguageOption).toHaveBeenCalledTimes(1);
    expect(mockSetSelectedLanguageOption).toHaveBeenCalledWith(valueFromEvent);
  });
  it('calls correct callback for route selector', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root.findAllByProps({ inputLabel: 'Route' })[0].props.onChange(event);

    expect(mockSetSelectedRouteOption).toHaveBeenCalledTimes(1);
    expect(mockSetSelectedRouteOption).toHaveBeenCalledWith(valueFromEvent);
  });
  it('calls correct callback for auto pay customers checkbox', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ label: 'Autopay customers' })[0]
      .props.onChange(event);

    expect(mockSetIsAutopayCustomersOption).toHaveBeenCalledTimes(1);
    expect(mockSetIsAutopayCustomersOption).toHaveBeenCalledWith(true);
  });
  it('calls correct callback for communication type selector', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ inputLabel: 'Communication type' })[0]
      .props.onChange(event);

    expect(mockSetSelectedCommunicationTypeOption).toHaveBeenCalledTimes(1);
    expect(mockSetSelectedCommunicationTypeOption).toHaveBeenCalledWith(
      valueFromEvent
    );
  });
  it('calls correct callback for due date all checkbox', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root.findAllByProps({ label: 'All' })[0].props.onChange(event);

    expect(mockOnDueDateCheckboxChanged).toHaveBeenCalledTimes(1);
    expect(mockOnDueDateCheckboxChanged).toHaveBeenCalledWith(event);
  });
  it('calls correct callback for due date from value change', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ fromLabel: 'From' })[0]
      .props.onFromChanged('new-from');

    expect(mockOnDueDateFromOptionChanged).toHaveBeenCalledTimes(1);
    expect(mockOnDueDateFromOptionChanged).toHaveBeenCalledWith('new-from');
  });
  it('calls correct callback for due date to value change', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root.findAllByProps({ toLabel: 'To' })[0].props.onToChanged('new-to');

    expect(mockOnDueDateToOptionChanged).toHaveBeenCalledTimes(1);
    expect(mockOnDueDateToOptionChanged).toHaveBeenCalledWith('new-to');
  });
  it('calls correct callback for Customers With No Call Activity selector', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root
      .findAllByProps({ inputLabel: 'Customer’s Activity' })[0]
      .props.onChange(event);

    expect(mockOnCustomersWithActivityChanged).toHaveBeenCalledTimes(1);
    expect(mockOnCustomersWithActivityChanged).toHaveBeenCalledWith(
      valueFromEvent
    );
  });
  it('calls correct callback for Clear button', () => {
    const tree = renderer.create(
      <Filters handleToggleFilterPanel={() => {}} isOpen />
    );

    tree.root.findAllByProps({ variant: 'text' })[0].props.onClick(event);

    expect(mockOnClearSelectedOptions).toHaveBeenCalledTimes(1);
  });
});
