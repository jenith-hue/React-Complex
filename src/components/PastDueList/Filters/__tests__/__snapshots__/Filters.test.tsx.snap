// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters renders inputs with values from context 1`] = `
<div
  className="filtersRoot"
>
  <div
    className="card"
    data-testid="RACCard"
  >
    <div
      className="cardBody"
    >
      <div
        className="row"
      >
        <div
          container={true}
          data-testid="Grid"
          spacing={3}
        >
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Days late 1",
                ]
              }
              inputLabel="Days late"
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Days late 1",
                    "referenceCode": "1",
                    "value": "Days late 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Best time to call 1",
                ]
              }
              inputLabel="Best Time to Call"
              loading={false}
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Best time to call 1",
                    "referenceCode": "1",
                    "value": "Best time to call 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Commitment status 1",
                ]
              }
              inputLabel="Commitment Status"
              loading={false}
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Commitment status 1",
                    "referenceCode": "1",
                    "value": "Commitment status 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Language option 1",
                ]
              }
              inputLabel="Language"
              loading={false}
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Language option 1",
                    "referenceCode": "1",
                    "value": "Language option 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              classes={
                Object {
                  "paper": "selectOptionsPaper",
                }
              }
              data-testid="RACSelect"
              defaultValue="Route 1"
              inputLabel="Route"
              loading={false}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Route 1",
                    "routeCode": "1",
                    "routeDescription": "1",
                    "value": "Route 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              checked={true}
              classes={
                Object {
                  "checkbox": "noPaddingLeft",
                }
              }
              data-testid="RACCheckBox"
              label="Autopay customers"
              onChange={[Function]}
              size="medium"
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Communication type 1",
                ]
              }
              inputLabel="Communication type"
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Communication type 1",
                    "referenceCode": "1",
                    "value": "Communication type 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue="Customers with no call activity 1"
              infoTitle="Customers with no call activity 1"
              inputLabel="Customerâ€™s Activity"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Customers with  activity 1",
                    "referenceCode": "1",
                    "value": "Customers with activity 1",
                  },
                ]
              }
              optionsToolTips={
                Array [
                  Object {
                    "label": "Customer with and without activity",
                    "value": "All",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={6}
            md={7}
            sm={12}
            xs={12}
          >
            <div
              data-testid="Divider"
              flexItem={true}
              orientation="vertical"
            />
            <div
              className="centerVertically"
            >
              <div
                className="multipleElementsInGridTitle"
                data-testid="Typography"
                variant="body1"
              >
                Due date
              </div>
              <div
                checked={true}
                classes={
                  Object {
                    "checkbox": "noPaddingLeft",
                  }
                }
                data-testid="RACCheckBox"
                label="All"
                onChange={[MockFunction]}
                size="medium"
              />
            </div>
            <div
              className="multipleElementsInGrid"
            >
              <div
                className=""
              >
                <div
                  data-testid="RACDateRangePicker"
                  disabled={true}
                  fromErrorMessage=""
                  fromLabel="From"
                  fromValue="Due date from 1"
                  minTo="Due date from 1"
                  name="dueDate"
                  onFromChanged={[Function]}
                  onToChanged={[Function]}
                  toErrorMessage=""
                  toLabel="To"
                  toValue="Due date to 1"
                />
              </div>
            </div>
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={6}
            md={6}
            sm={12}
            xs={12}
          >
            <div
              data-testid="Divider"
              flexItem={true}
              orientation="vertical"
            />
            <div
              className="centerVertically"
            >
              <div
                className="multipleElementsInGridTitle"
                data-testid="Typography"
                variant="body1"
              >
                Commitment date
              </div>
              <div
                checked={true}
                classes={
                  Object {
                    "checkbox": "noPaddingLeft",
                  }
                }
                data-testid="RACCheckBox"
                label="All"
                size="medium"
              />
            </div>
            <div
              className="multipleElementsInGrid"
            >
              <div>
                <div
                  data-testid="RACDateRangePicker"
                  disabled={true}
                  fromErrorMessage=""
                  fromLabel="From"
                  fromValue="Commitment date from 1"
                  name="commitmentDateRange"
                  onFromChanged={[Function]}
                  onToChanged={[Function]}
                  toErrorMessage=""
                  toLabel="To"
                  toValue="Commitment date to 1"
                />
              </div>
            </div>
          </div>
          <div
            classes={
              Object {
                "root": "buttonContainer",
              }
            }
            data-testid="Grid"
            item={true}
            lg={12}
            md={12}
            sm={12}
            xs={12}
          >
            <div
              color="primary"
              data-testid="RACButton"
              onClick={[Function]}
              size="large"
              variant="text"
            >
              Clear
            </div>
            <div
              color="primary"
              data-testid="RACButton"
              onClick={[Function]}
              size="large"
              variant="contained"
            >
              Apply
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters renders inputs with values from context, even when filters are not open 1`] = `
<div
  className="filtersRoot"
>
  <div
    className="card"
    data-testid="RACCard"
  >
    <div
      className="cardBody"
    >
      <div
        className="row"
      >
        <div
          container={true}
          data-testid="Grid"
          spacing={3}
        >
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Days late 1",
                ]
              }
              inputLabel="Days late"
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Days late 1",
                    "referenceCode": "1",
                    "value": "Days late 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Best time to call 1",
                ]
              }
              inputLabel="Best Time to Call"
              loading={false}
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Best time to call 1",
                    "referenceCode": "1",
                    "value": "Best time to call 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Commitment status 1",
                ]
              }
              inputLabel="Commitment Status"
              loading={false}
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Commitment status 1",
                    "referenceCode": "1",
                    "value": "Commitment status 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Language option 1",
                ]
              }
              inputLabel="Language"
              loading={false}
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Language option 1",
                    "referenceCode": "1",
                    "value": "Language option 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              classes={
                Object {
                  "paper": "selectOptionsPaper",
                }
              }
              data-testid="RACSelect"
              defaultValue="Route 1"
              inputLabel="Route"
              loading={false}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Route 1",
                    "routeCode": "1",
                    "routeDescription": "1",
                    "value": "Route 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              checked={true}
              classes={
                Object {
                  "checkbox": "noPaddingLeft",
                }
              }
              data-testid="RACCheckBox"
              label="Autopay customers"
              onChange={[Function]}
              size="medium"
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue={
                Array [
                  "Communication type 1",
                ]
              }
              inputLabel="Communication type"
              multiple={true}
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Communication type 1",
                    "referenceCode": "1",
                    "value": "Communication type 1",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={2}
            md={3}
            sm={3}
            xs={6}
          >
            <div
              data-testid="RACSelect"
              defaultValue="Customers with no call activity 1"
              infoTitle="Customers with no call activity 1"
              inputLabel="Customerâ€™s Activity"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "defaultValue": "1",
                    "description": "1",
                    "displaySeq": 1,
                    "label": "Customers with  activity 1",
                    "referenceCode": "1",
                    "value": "Customers with activity 1",
                  },
                ]
              }
              optionsToolTips={
                Array [
                  Object {
                    "label": "Customer with and without activity",
                    "value": "All",
                  },
                ]
              }
            />
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={6}
            md={7}
            sm={12}
            xs={12}
          >
            <div
              data-testid="Divider"
              flexItem={true}
              orientation="vertical"
            />
            <div
              className="centerVertically"
            >
              <div
                className="multipleElementsInGridTitle"
                data-testid="Typography"
                variant="body1"
              >
                Due date
              </div>
              <div
                checked={true}
                classes={
                  Object {
                    "checkbox": "noPaddingLeft",
                  }
                }
                data-testid="RACCheckBox"
                label="All"
                onChange={[MockFunction]}
                size="medium"
              />
            </div>
            <div
              className="multipleElementsInGrid"
            >
              <div
                className=""
              >
                <div
                  data-testid="RACDateRangePicker"
                  disabled={true}
                  fromErrorMessage=""
                  fromLabel="From"
                  fromValue="Due date from 1"
                  minTo="Due date from 1"
                  name="dueDate"
                  onFromChanged={[Function]}
                  onToChanged={[Function]}
                  toErrorMessage=""
                  toLabel="To"
                  toValue="Due date to 1"
                />
              </div>
            </div>
          </div>
          <div
            classes={
              Object {
                "root": "grid",
              }
            }
            data-testid="Grid"
            item={true}
            lg={6}
            md={6}
            sm={12}
            xs={12}
          >
            <div
              data-testid="Divider"
              flexItem={true}
              orientation="vertical"
            />
            <div
              className="centerVertically"
            >
              <div
                className="multipleElementsInGridTitle"
                data-testid="Typography"
                variant="body1"
              >
                Commitment date
              </div>
              <div
                checked={true}
                classes={
                  Object {
                    "checkbox": "noPaddingLeft",
                  }
                }
                data-testid="RACCheckBox"
                label="All"
                size="medium"
              />
            </div>
            <div
              className="multipleElementsInGrid"
            >
              <div>
                <div
                  data-testid="RACDateRangePicker"
                  disabled={true}
                  fromErrorMessage=""
                  fromLabel="From"
                  fromValue="Commitment date from 1"
                  name="commitmentDateRange"
                  onFromChanged={[Function]}
                  onToChanged={[Function]}
                  toErrorMessage=""
                  toLabel="To"
                  toValue="Commitment date to 1"
                />
              </div>
            </div>
          </div>
          <div
            classes={
              Object {
                "root": "buttonContainer",
              }
            }
            data-testid="Grid"
            item={true}
            lg={12}
            md={12}
            sm={12}
            xs={12}
          >
            <div
              color="primary"
              data-testid="RACButton"
              onClick={[Function]}
              size="large"
              variant="text"
            >
              Clear
            </div>
            <div
              color="primary"
              data-testid="RACButton"
              onClick={[Function]}
              size="large"
              variant="contained"
            >
              Apply
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
